### PR TITLE
model: support database placement option and partition placement option

### PR DESCRIPTION
**Pull request by zhaoxugang**
### What problem does this PR solve? 
support database placement option and partition placement option

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by zhaoxugang** [comment]
PTAL @AilinKid @xhebox

File: model/ddl.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ const (
 	ActionCreatePlacementPolicy         ActionType = 51
 	ActionAlterPlacementPolicy          ActionType = 52
 	ActionDropPlacementPolicy           ActionType = 53
+	ActionAlterTablePartitionOptions    ActionType = 54
```
Comment (xhebox) [inline model/ddl.go:87]: Maybe a smaller name. We will transform `ast.TableOption` into different actions, refer https://github.com/pingcap/tidb/blob/257140a154a13061b55993f499aa25f0cd07cdf6/ddl/ddl_api.go#L2530-L2573. We don't really have `AlterTableOption` type.

**Review by xhebox — COMMENTED** [review]
Overlap with #1340, looks like. Maybe you need to discuss with @mjonss .

**Comment by mjonss** [comment]
I just merged https://github.com/pingcap/parser/pull/1340 so the model now has both `PlacementPolicyRef` and `DirectPlacementOpts` on both `DBInfo` and `PartitionDefinition`.

File: model/ddl.go (hunk 1 above)
Comment (zhaoxugang) [reply model/ddl.go:87]: > Maybe a smaller name. We will transform `ast.TableOption` into different actions, refer https://github.com/pingcap/tidb/blob/257140a154a13061b55993f499aa25f0cd07cdf6/ddl/ddl_api.go#L2530-L2573. We don't really have `AlterTableOption` type.

the constant is used for ddl.runDDLJob() to mark job type not ddl.AlterTable()

**Review by xhebox — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Pull request merged**